tests/conftest.py

```diff
@@ -12,26 +12,70 @@
 from workos.utils.request_helper import DEFAULT_LIST_RESPONSE_LIMIT
 
 
-@pytest.fixture
-def sync_http_client_for_test():
-    return SyncHTTPClient(
+_TEST_HTTP_CLIENTS = {
+    "sync": SyncHTTPClient(
         api_key="sk_test",
         base_url="https://api.workos.test/",
         client_id="client_b27needthisforssotemxo",
         version="test",
-    )
-
-
-@pytest.fixture
-def async_http_client_for_test():
-    return AsyncHTTPClient(
+    ),
+    "async": AsyncHTTPClient(
         api_key="sk_test",
         base_url="https://api.workos.test/",
         client_id="client_b27needthisforssotemxo",
         version="test",
+    ),
+}
+
+
+def pytest_configure(config) -> None:
+    config.addinivalue_line(
+        "markers", "sync_and_async(): mark test to run both sync and async versions"
```

Comment:
In the end, we're not actually running async tests, we're just testing multiple clients/modules, right? How different is this from just supplying a list or tuple of modules rather than naming?

Reply:
Without some sort of sync and async parent class, I'm not sure there's a great way to do this.

Reply:
Parent class, or `Protocol`? What's the need?

Reply:
We need to know if the `http_client` on the module should be `sync` or `async`. The `Protocol` allows either.

Reply:
... could the http client be passed in with the annotation string? Is that crazy?